lib/Cake/Network/Email/CakeEmail.php

```diff
@@ -1717,6 +1721,27 @@ protected function _renderTemplates($content) {
 			$rendered[$type] = implode("\n", $rendered[$type]);
 			$rendered[$type] = rtrim($rendered[$type], "\n");
 		}
+
+		/* Embed images inline in html templates */
+		if (!empty($rendered['html'])) {
```

Comment:
Other test, more real:
```
$loops = 10000000;
$t1 = explode(' ', microtime());
for ($i = 0; $i < $loops; $i++) {
	preg_match_all('~<img[^>]*src\s*=\s*(["\'])(cid://|file://|cid:|file:)([^\1]+)\1~iU', print_r($this->viewVars, true), $userFiles);
	$userFiles = array_unique($userFiles[3]);
	preg_match_all('~<img[^>]*src\s*=\s*(["\'])(cid://|file://|cid:|file:)([^\1]+)\1~iU', $rendered['html'], $embebFiles);
	$embebFiles = array_unique($embebFiles[3]);
	$embebFiles = array_diff($embebFiles, $userFiles);
}
$t2 = explode(' ', microtime());
$t = $t2[1] - $t1[1];
$t += $t2[0] - $t1[0];
debug($embebFiles);
debug("print_r:\t" . $t);

$t1 = explode(' ', microtime());
for ($i = 0; $i < $loops; $i++) {
	preg_match_all('~<img[^>]*src\s*=\s*(["\'])(cid://|file://|cid:|file:)([^\1]+)\1~iU', serialize($this->viewVars), $userFiles);
	$userFiles = array_unique($userFiles[3]);
	preg_match_all('~<img[^>]*src\s*=\s*(["\'])(cid://|file://|cid:|file:)([^\1]+)\1~iU', $rendered['html'], $embebFiles);
	$embebFiles = array_unique($embebFiles[3]);
	$embebFiles = array_diff($embebFiles, $userFiles);
}
$t2 = explode(' ', microtime());
$t = $t2[1] - $t1[1];
$t += $t2[0] - $t1[0];
debug($embebFiles);
debug("serialize:\t" . $t);

```
print_r:       137.821467 seconds
serialize:     112.265272 seconds

If we include all the rendering process templates will be shown that the price of serializing just change the result in terms of performance.
Likewise the two preg_replace not alter the overall performance.